timely/src/dataflow/operators/to_stream.rs

```diff
@@ -56,3 +63,215 @@ impl<T: Timestamp, I: IntoIterator+'static> ToStream<T, I::Item> for I where I::
         })
     }
 }
+
+/// Errors encountered during event streaming
+#[derive(Clone, Copy, Debug)]
+pub enum StreamError {
+    /// Encountered time less than the most recent Progress event
+    TimeMovedBackwards,
+    /// The error returned when activation fails across thread boundaries because the receiving end
+    /// has hung up.
+    SyncActivationError,
+}
+
+/// Data and progress events of the native stream.
+pub enum Event<T, D> {
+    /// Indicates that timestamps have advanced to time T
+    Progress(T),
+    /// Indicates that event D happened at time T
+    Message(T, D),
+}
```

Comment:
We should talk through this! There is already a similar enum in the `capture` module, and it is a bit more rich than this. In particular, this enum isn't able to capture partially ordered times, where there may be a set of capabilities rather than just one.

Reply:
I did see that one but wasn't sure what the Vec is for. Let's talk about it tomorrow